# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Navigator.h
# Commit: f50ff29a1a64
# Full Hash: f50ff29a1a649e3aad723b592bdd2c3c325b3d99
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2023-12-06 05:18:37
# Regressor Bug: 1589554
# File Overlap Count: 1
# Description:
#   Bug 1589554 - Part 1: Screen Wake Lock boilerplate. r=dom-core,smaug,emilio,edgar
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D189508
# ==============================================================================

diff -r 00afe2e46b1b -r f50ff29a1a64 dom/base/Navigator.h
--- a/dom/base/Navigator.h	Tue Dec 05 23:29:16 2023 +0000
+++ b/dom/base/Navigator.h	Tue Dec 05 23:58:06 2023 +0000
@@ -45,6 +45,7 @@
 class LockManager;
 class HTMLMediaElement;
 class AudioContext;
+class WakeLockJS;
 }  // namespace dom
 namespace webgpu {
 class Instance;
@@ -129,6 +130,7 @@
   bool GlobalPrivacyControl();
   Geolocation* GetGeolocation(ErrorResult& aRv);
   Promise* GetBattery(ErrorResult& aRv);
+  dom::WakeLockJS* WakeLock();
 
   bool CanShare(const ShareData& aData);
   already_AddRefed<Promise> Share(const ShareData& aData, ErrorResult& aRv);
@@ -300,6 +302,7 @@
   RefPtr<Promise> mSharePromise;  // Web Share API related
   RefPtr<dom::LockManager> mLocks;
   RefPtr<dom::UserActivation> mUserActivation;
+  RefPtr<dom::WakeLockJS> mWakeLock;
 };
 
 }  // namespace mozilla::dom